### PR TITLE
Generate common files across taipy repos

### PR DESCRIPTION
**Pull request by AaryaZ**
This PR addresses Issue #106.
Any reviewed changes will be immediately addressed.
This PR is raised against the requested changes suggested in https://github.com/Avaiga/taipy/pull/358 .
Faced a few issues while making changes to the existing PR, which is why i created New PR.

**Requirement:** Any push or any change in any of the common files on taipy would automatically be pushed across the relevant repositories (namely taipy-config, taipy-core, taipy-gui, taipy-rest).

**Approach:**
To meet this requirement, `Repo File Sync Action` is used.
For more details, visit this [link](https://github.com/marketplace/actions/repo-file-sync-action#%EF%B8%8F-sync-configuration).

PS: In order for the Action to access your repositories you have to specify a Personal Access token as the value for GH_PAT.(GITHUB_TOKEN will not work). [The PAT needs the full repo scope. ](https://github.com/BetaHuhn/repo-file-sync-action/discussions/31#discussioncomment-674804)

### CODE REVIEW

**Comment by AaryaZ** [comment]
@trgiangdo @jrobinAV pls review this PR under `hacktoberfest`.

File: .github/sync.yml
Comment (trgiangdo) [inline .github/sync.yml:1]: This file should be inside the `worlflows` folder

File: .github/sync.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    - .gitattributes
+    - .license-header
+    - CODE_OF_CONDUCT.md
+    - CONTRIBUTING.md
+    - LICENSE
+    - mypy.ini
+    - pyproject.toml
+    - src/taipy/__init__.py
+    # - .isort.cfg
+    # - src/__init__.py
```
Comment (trgiangdo) [inline .github/sync.yml]:
```suggestion
    # - src/__init__.py

```
missing EOL

File: .github/workflows/sync.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@master
+      - name: Run GitHub File Sync
+        uses: BetaHuhn/repo-file-sync-action@v1
+        with:
+          GH_PAT: ${{ secrets.GH_PAT }}
+
+#In order for the Action to access your repositories you have to specify a Personal Access token as the value for GH_PAT (GITHUB_TOKEN will not work). [The PAT needs the full repo scope . ](https://github.com/BetaHuhn/repo-file-sync-action/discussions/31#discussioncomment-674804)
```
Comment (trgiangdo) [inline .github/workflows/sync.yml]:
```suggestion
```
This comment is not necessary on the file. Please put it as a comment on the PR.

File: .github/sync.yml
Comment (AaryaZ) [reply .github/sync.yml:1]: Actually the [article](https://github.com/marketplace/actions/repo-file-sync-action#%EF%B8%8F-sync-configuration) suggests creating sync.yml file in .github folder. A snippet of the article directing this is given below.
image


Will shift the file in workflows folder if that is how it is supposed to be.

Comment (trgiangdo) [reply .github/sync.yml:1]: Oh, that's my bad. Looks good then :D Thank you

**Comment by AaryaZ** [comment]
@trgiangdo Are there any other changes to be done?

**Comment by trgiangdo** [comment]
I'm asking for the team to provide more feedback

File: .github/workflows/sync.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+name: Sync Files
+on:
+  push:
+    branches:
+      - master
```
Comment (jrobinAV) [inline .github/workflows/sync.yml]: Our main branch is called develop.

Comment (AaryaZ) [reply .github/workflows/sync.yml]: Done

**Review by jrobinAV — DISMISSED** [review]

File: .github/workflows/sync.yml [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  push:
+    branches:
+      - develop
+  workflow_dispatch:
+jobs:
+  sync:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@master
```
Comment (trgiangdo) [inline .github/workflows/sync.yml]:
```suggestion
        uses: actions/checkout@develop
```

**Review by trgiangdo — DISMISSED** [review]
LGTM for now. We can merge the PR and improve the action latter when the token is added to test it out.

Thank you a lot @AaryaZ for your contribution

File: .github/workflows/sync.yml (hunk 4 above)
Comment (AaryaZ) [reply .github/workflows/sync.yml]: Thanks. Done.

**Comment by AaryaZ** [comment]
@trgiangdo Thank You.

**Review by trgiangdo — APPROVED** [review]

**Comment by jrobinAV** [comment]
Thank you for your contribution!

**Comment by joaoandre-avaiga** [comment]
LGTM! Thanks @AaryaZ !

**Review by AaryaZ — COMMENTED** [review]
@joaoandre-avaiga @jrobinAV @trgiangdo Thanks. Eagerly Waiting for PR to get merge.

**Comment by marisogo** [comment]
Hi @AaryaZ, could you send me an e-mail at  marine.gosselin@taipy.io so we can send you some swag!

**Comment by AaryaZ** [comment]
@marisogo Thank you that is really so sweet. Sent a mail to the given mail address.
My email: aaryazunjarrao@gmail.com

**Pull request merged**